src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java

```diff
@@ -89,14 +89,15 @@ public CoverageProcessor(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @N
      * @param reportAdapters   reportAdapters specified by user
      * @param reportDetectors  reportDetectors specified by user
      * @param globalThresholds global threshold specified by user
+     * @return coverage action containing aggregated reports from adaptors or null
      */
-    public void performCoverageReport(List<CoverageReportAdapter> reportAdapters, List<ReportDetector> reportDetectors, List<Threshold> globalThresholds)
+    public CoverageAction performCoverageReport(List<CoverageReportAdapter> reportAdapters, List<ReportDetector> reportDetectors, List<Threshold> globalThresholds)
             throws IOException, InterruptedException, CoverageException {
         Map<CoverageReportAdapter, List<CoverageResult>> results = convertToResults(reportAdapters, reportDetectors);
 
         CoverageResult coverageReport = aggregateReports(results);
         if (coverageReport == null) {
-            return;
+            return null;
```

Comment:
This looks somewhat strange (but this is from the existing code). When does the report 
- return `null`
- throw an `CoverageException`

Reply:
I refactored my code to get `CoverageAction` directly from `run` instead of changing this `CoverageProcessor`.